File: MB-Secretaria/review/review-audit.ts
import type { ClinicalReviewCase, ReviewAuditEvent, ReviewDecision, ReviewStatus } from "./types";

export function createReviewAuditEvent(input: {
  reviewCase: ClinicalReviewCase;
  status_before?: ReviewStatus;
  status_after: ReviewStatus;
  decision?: ReviewDecision;
  blocked?: boolean;
}): ReviewAuditEvent {
  return {
    review_id: input.reviewCase.review_id,
    trace_id: input.reviewCase.trace_id,
    reviewer_id: input.decision?.reviewer_id,
    status_before: input.status_before,
    status_after: input.status_after,
    escalation_level: input.reviewCase.escalation_level,
    override_used: input.decision?.decision === "override",
    blocked: input.blocked ?? input.status_after === "AUTO_BLOCKED",
    created_at: new Date().toISOString(),
  };
}

export class InMemoryReviewAuditSink {
  private events: ReviewAuditEvent[] = [];

  append(event: ReviewAuditEvent): void {
    this.events.push(event);
  }

  list(): ReviewAuditEvent[] {
    return [...this.events];
  }
}

File: MetaBrain/review_py/review_audit.py
from __future__ import annotations

from datetime import UTC, datetime

from .types import ClinicalReviewCase, ReviewAuditEvent, ReviewDecision, ReviewStatus


def create_review_audit_event(
    review_case: ClinicalReviewCase,
    status_after: ReviewStatus,
    status_before: ReviewStatus | None = None,
    decision: ReviewDecision | None = None,
    blocked: bool | None = None,
) -> ReviewAuditEvent:
    return ReviewAuditEvent(
        review_id=review_case.review_id,
        trace_id=review_case.trace_id,
        reviewer_id=decision.reviewer_id if decision else None,
        status_before=status_before,
        status_after=status_after,
        escalation_level=review_case.escalation_level,
        override_used=bool(decision and decision.decision == "override"),
        blocked=bool(blocked if blocked is not None else status_after == "AUTO_BLOCKED"),
        created_at=datetime.now(UTC).isoformat(),
    )


class InMemoryReviewAuditSink:
    def __init__(self) -> None:
        self._events: list[ReviewAuditEvent] = []

    def append(self, event: ReviewAuditEvent) -> None:
        self._events.append(event)

    def list(self) -> list[ReviewAuditEvent]:
        return list(self._events)
